presto-salesforce/src/main/java/io/prestosql/plugin/salesforce/driver/connection/ForceConnection.java

```diff
@@ -0,0 +1,483 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.salesforce.driver.connection;
+
+import com.sforce.soap.partner.PartnerConnection;
+import io.prestosql.plugin.salesforce.driver.metadata.ForceDatabaseMetaData;
+import io.prestosql.plugin.salesforce.driver.metadata.ForceDatabaseMetadataCache;
+import io.prestosql.plugin.salesforce.driver.statement.ForcePreparedStatement;
+
+import javax.inject.Inject;
+import javax.inject.Provider;
+
+import java.sql.Array;
+import java.sql.Blob;
+import java.sql.CallableStatement;
+import java.sql.Clob;
+import java.sql.Connection;
+import java.sql.DatabaseMetaData;
+import java.sql.NClob;
+import java.sql.PreparedStatement;
+import java.sql.SQLClientInfoException;
+import java.sql.SQLException;
+import java.sql.SQLWarning;
+import java.sql.SQLXML;
+import java.sql.Savepoint;
+import java.sql.Statement;
+import java.sql.Struct;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+import java.util.concurrent.Executor;
+import java.util.logging.Logger;
+
+public class ForceConnection
+        implements Connection
+{
+    private static final String SF_JDBC_DRIVER_NAME = "SF JDBC driver";
+    @Inject @ForceDatabaseMetadataCache private static Provider<Map<String, ForceDatabaseMetaData>> metadataCacheProvider;
```

Comment:
Can we maintain this cache in `PrestoDriver` and pass it each time we create .a new connection

Reply:
I need to refactor this whole thing, since right now the metadata cache does not expire.